iij/ngx_http_secure_token_iijpta.c

```diff
@@ -0,0 +1,191 @@
+#include "ngx_http_secure_token_iijpta.h"
+#include "../ngx_http_secure_token_filter_module.h"
+#include "../ngx_http_secure_token_utils.h"
+
+#include <time.h>
+#include <openssl/hmac.h>
+#include <openssl/evp.h>
+
+// constants
+#define CRC32_SIZE    4
+#define DEADLINE_SIZE 8
+#define PATH_LIMIT    1024
+	
+// typedefs
+typedef struct {
+	ngx_str_t key;
+	ngx_str_t iv;
+	ngx_str_t path;
+	time_t    timelimit;
+} ngx_secure_token_iijpta_token_t;
+
+// globals
+static ngx_command_t ngx_http_secure_token_iijpta_cmds[] = {
+	{ ngx_string("key"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, key),
+	NULL },
+	
+	{ ngx_string("iv"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, iv),
+	NULL },
+
+	{ ngx_string("path"),
+	NGX_CONF_TAKE1,
+	ngx_conf_set_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, path),
+	NULL },
+
+	{ ngx_string("timelimit"),
+	NGX_CONF_TAKE1,
+	ngx_conf_set_sec_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, timelimit),
+	NULL },
+};
+
+static ngx_int_t
+ngx_secure_token_iijpta_get_var(
+	ngx_http_request_t *r,
+	ngx_http_variable_value_t *v,
+	uintptr_t data)
+{
+	EVP_CIPHER_CTX ctx;
+	uint32_t crc;
+	ngx_secure_token_iijpta_token_t* token = (void*)data;
+	u_char* in;
+	size_t in_len  = CRC32_SIZE + DEADLINE_SIZE + token->path.len;
+	u_char *qparam;
```

Comment:
usually in nginx, such moving pointers are named `p`

Reply:
fix at 582635f